datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/load/AspectResolver.java

```diff
@@ -0,0 +1,44 @@
+package com.linkedin.datahub.graphql.resolvers.load;
+
+import com.linkedin.datahub.graphql.AspectLoadKey;
+import com.linkedin.datahub.graphql.generated.Aspect;
+import com.linkedin.datahub.graphql.generated.Entity;
+import com.linkedin.datahub.graphql.resolvers.ResolverUtils;
+import com.linkedin.datahub.graphql.types.aspect.AspectMapper;
+import com.linkedin.metadata.aspect.AspectWithMetadata;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.concurrent.CompletableFuture;
+import org.dataloader.DataLoader;
+
+
+/**
+ * Generic GraphQL resolver responsible for
+ *
+ *    1. Generating a single input AspectLoadKey.
+ *    2. Resolving a single {@link Aspect}.
+ *
+ */
+public class AspectResolver implements DataFetcher<CompletableFuture<Aspect>> {
+
+    public AspectResolver() {
+    }
+
+    @Override
+    public CompletableFuture<Aspect> get(DataFetchingEnvironment environment) {
+        final DataLoader<AspectLoadKey, Aspect> loader = environment.getDataLoaderRegistry().getDataLoader("Aspect");
+
+        String fieldName = environment.getField().getName();
+        Long version = environment.getArgument("version");
+        String urn = ((Entity) environment.getSource()).getUrn();
+
+        // first, we try fetching the aspect from the local cache
+        AspectWithMetadata aspectFromContext = ResolverUtils.getAspectFromLocalContext(environment);
```

Comment:
We are positive this results in no downstream calls in the common case?

Reply:
Yep!